lstchain/tools/lstchain_create_dl2_file.py

```diff
@@ -0,0 +1,200 @@
+"""
+Create h5 file with reconstruction of energy, disp and gamma/hadron separation of events.
+"""
+
+import joblib
+import numpy as np
+import pandas as pd
+from ctapipe.core import Provenance, Tool, ToolConfigurationError, traits
+from lstchain.io import get_dataset_keys, write_dl2_dataframe
+from lstchain.io.io import (dl1_images_lstcam_key, dl1_params_lstcam_key,
+                            dl1_params_src_dep_lstcam_key,
+                            dl2_params_src_dep_lstcam_key, write_dataframe)
+from lstchain.reco import dl1_to_dl2
+from lstchain.reco.utils import add_delta_t_key, filter_events, impute_pointing
+from tables import open_file
+
+
+class ReconstructionHDF5Writer(Tool):
+    name = "ReconstructionHDF5Writer"
+    description = "Generate a HDF5 file with reconstructed energy, disp and gammaness of events"
+
+    input = traits.Path(help="Path to a DL1 HDF5 file", directory_ok=False, exists=True).tag(config=True)
+    path_energy_model = traits.Path(
+        help="Path where to find the Energy trained RF model file",
+        directory_ok=False,
+        exists=True,
+    ).tag(config=True)
+    path_disp_model = traits.Path(
+        help="Path where to find the Disp trained RF model file",
+        directory_ok=False,
+        exists=True,
+    ).tag(config=True)
+    path_gh_model = traits.Path(
+        help="Path where to find the Gammaness trained RF model file",
+        directory_ok=False,
+        exists=True,
+    ).tag(config=True)
+    source_dependent = traits.Bool(
+        default_value=False,
+        help="Is the analysis source dependent?"
+    ).tag(config=True)
+    events_filters = traits.Dict(
+        help="Dictionary with information to filter events"
+    ).tag(config=True)
+    classification_features = traits.List(
+        help="List of classification features"
+    ).tag(config=True)
+    regression_features = traits.List(
+        help="List of regression features"
+    ).tag(config=True)
+    output_dir = traits.Path(help="Path where to store the reconstructed DL2", file_ok=False).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "ReconstructionHDF5Writer.input",
+        ("o", "output"): "ReconstructionHDF5Writer.output_dir",
+        "energy-model": "ReconstructionHDF5Writer.path_energy_model",
+        "disp-model": "ReconstructionHDF5Writer.path_disp_model",
+        "gh-model": "ReconstructionHDF5Writer.path_gh_model",
+    }
+
+    flags = {
+        'source-dependent': (
+            {'ReconstructionHDF5Writer': {'source_dependent': True}},
+            'Perform source dependent analysis',
+        )
+    }
+
+    def __init__(self, **kwargs):
```

Comment:
If `__init__` does nothing, it is not needed.

Reply:
ok, I have removed it